### PR TITLE
Travis CI: Add Python 3.8 and flake8 to testing

### PR DESCRIPTION
**Pull request by cclauss**
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -112,13 +108,22 @@ jobs:
       name: Python Style and Linter
       <<: *stage_linux
       script:
+        - pip install flake8
```
Comment (mtreinish) [inline .travis.yml]: We should just add this to the requirements-dev.txt file.

File: .travis.yml [hunk 2]
```diff
@@ -112,13 +108,22 @@ jobs:
...
+        - flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (mtreinish) [inline .travis.yml]: We should add this to the tox.ini too so people can run things locally.

**Comment by chriseclectic** [comment]
@cclauss @mtreinish Is there anything left to do here now that #428 has been merged?

**Comment by cclauss** [comment]
Sure...  Add the flake8 tests and fix the three undefined name...

flake8 testing of https://github.com/Qiskit/qiskit-aer on Python 3.8.0

$ __flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics__
```
./test/terra/utils/mock.py:248:15: F821 undefined name 'FakeQasmSimulator'
    backend = FakeQasmSimulator()
              ^
./test/benchmark/fusion_benchmarks.py:55:19: F821 undefined name 'QiskitError'
            raise QiskitError("Simulation failed. Status: " + result.status)
                  ^
./test/benchmark/fusion_benchmarks.py:110:19: F821 undefined name 'QiskitError'
            raise QiskitError("Simulation failed. Status: " + result.status)                  ^
3     F821 undefined name 'QiskitError'
3
```

**Comment by cclauss** [comment]
I am not sure how to fix this first one because __ FakeQasmSimulator__ is [defined in qiskit-terra](https://github.com/Qiskit/qiskit-terra/search?q=FakeQasmSimulator&unscoped_q=FakeQasmSimulator) so I am not clear how to import it.

**Pull request closed without merging**